### DISCUSSION TITLE
How to load net module without autosetup

### DISCUSSION THREAD

**Discussion by albham** [Moddable-OpenSource/moddable#953] [Q&A] [answered]
Hello,

   I am starting with Moddable SDK on ESP32.

Looking at the examples (`wificonnection`) I see that you can load the `"wifi"` module and manage the connection 
by including `"$(MODDABLE)/modules/network/wifi/manifest.json"`.

But the only way to load the `"net"` module seems to be including `"$(MODDABLE)/examples/manifest_net.json"`, 
which also loads `setup/network` which tries to make the Wifi connection on startup.

So, I was wondering if there is a manifest to include that would allow access to "net" without the `setup/network`.

Thank you

Alberto

**Comment by phoddie** [comment, marked as answer]
> I am starting with Moddable SDK on ESP32.

Welcome!

It sounds like you want to manage the Wi-Fi connection yourself, instead of having the Moddable SDK auto-connect at start-up. There are two ways to do that:

1. Do not provide the `ssid` and `password` on the command line. To build, use a command line  like `mcconfig -d -m -p esp32` instead of `mcconfig -d -m -p esp32 ssid="foo" password=bar"`. Or, if you are defining the ssid and password in your project manifests `config` section, omit those. In both cases, `network/setup` will  see no credentials so it will not try to connect.
2. Remove the `setup/network` module from your project using exclude feature of the manifest (`~` ). In your projects manifest's `modules` section add:
```json
	"modules": {
		"~": [
			"$(BUILD)/devices/esp32/setup/network"
		],
		...
	}
```

> But the only way to load the "net" module seems to be including "$(MODDABLE)/examples/manifest_net.json",
which also loads setup/network which tries to make the Wifi connection on startup.

This isn't quite correct. Including "manifest_net.json"  is recommended as it is portable and maintained. The modules that need to include vary by platform and potentially over time as the implementation evolves. Including "manifest_net.json" should always work. But, you can merge the relevant information from "manifest_net.json" into your project's manifest instead, including only those modules that you need. This is more work than the two approaches described above.

**Reply by albham** [reply]
Hello,

  option 2 woks fine for me.

Thank you
